rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/ArtifactEndpoint.java

```diff
@@ -113,6 +117,43 @@
             @Parameter(description = FILTER_MD5_DESC) @QueryParam("md5") String md5,
             @Parameter(description = FILTER_SHA1_DESC) @QueryParam("sha1") String sha1);
 
+    static final String GET_ALL_FILTERED_DESC = "Gets all artifacts according to specified filters.";
+    static final String FILTER_IDENTIFIER_DESC = "Filter by artifact identifier or its part.";
+    static final String FILTER_QUALITY_DESC = "List of artifact qualities to include in result.";
+    static final String FILTER_REPOSITORY_TYPE_DESC = "Type of target repository.";
+
+    /**
+     * {@value GET_ALL_FILTERED_DESC}
+     *
+     * @param paginationParameters
+     * @param identifier {@value FILTER_IDENTIFIER_DESC}
+     * @param qualities {@value FILTER_QUALITY_DESC}
+     * @param repoType {@value FILTER_REPOSITORY_TYPE_DESC}
+     * @return
+     */
+    @Operation(
+            summary = GET_ALL_FILTERED_DESC,
+            responses = {
+                    @ApiResponse(
+                            responseCode = SUCCESS_CODE,
+                            description = SUCCESS_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ArtifactInfoPage.class))),
+                    @ApiResponse(
+                            responseCode = INVALID_CODE,
+                            description = INVALID_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = SERVER_ERROR_CODE,
+                            description = SERVER_ERROR_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @GET
+    @Path("/filter")
+    Page<ArtifactInfo> getAllFiltered(
+            @Valid @BeanParam PaginationParameters paginationParameters,
+            @Parameter(description = FILTER_IDENTIFIER_DESC) @QueryParam("identifier") String identifier,
+            @Parameter(description = FILTER_QUALITY_DESC) @QueryParam("qualities") Set<ArtifactQuality> qualities,
+            @Parameter(description = FILTER_REPOSITORY_TYPE_DESC) @QueryParam("repoType") RepositoryType repoType);
```

Comment:
Why are all the parameters optional? Wouldn't we always want to specify at least the identifier? And I would say repoType too.

Reply:
@pkocandr What parameters will be used at all times?

Reply:
@janinko Why to make some parameters optional? It is filtering, so it is up to the user/use-case. I'd keep all optional. It doesn't hurt and makes the usage more flexible.

Reply:
because of KISS

Reply:
I think that the Optionals do not make it somehow unreadable and the endpoint is more flexible this way. But if it sure that it won't be used any other way then it wouldn't make much sense to keep it optional.

Reply:
@janinko If it is optional, it is simple, isn't it? :-)

Reply:
For the version list read by DA we will specify all the params everytime, but for a different use-case in the future I would not limit it by making the params mandatory.

Reply:
Ok, I will leave them optional